pandas/tests/io/json/test_normalize.py

```diff
@@ -351,32 +353,35 @@ def test_non_ascii_key(self):
         ).decode("utf8")
 
         testdata = {
+            b"\xc3\x9cnic\xc3\xb8de".decode("utf8"): [0, 1],
             "sub.A": [1, 3],
             "sub.B": [2, 4],
-            b"\xc3\x9cnic\xc3\xb8de".decode("utf8"): [0, 1],
```

Comment:
Why is this change needed? (this input is a dict, not list of dicts, and plain dicts should not be impacted here (they also already preserve order))

Reply:
it definitely failed. My guess is json_normalize calls the list case on dataframe, which changes behavior.

Reply:
Ah, that makes sense. 
We might need to be careful here though. Do you know if the dicts from the json reader preserve the order of the file? Probably yes, but if not, we should still sort in this case to not have a random order.

Reply:
```
In [8]: pd.read_json(json.dumps([dict(a=1,b=2),dict(a=1,b=2, y=1, x=3),dict(b=1,a=2,z=4)]))
Out[8]: 
   a  b    y    x    z
0  1  2  NaN  NaN  NaN
1  1  2  1.0  3.0  NaN
2  2  1  NaN  NaN  4.0
```
I think that's ok. And the set of failing has been just these. If it were random, the many builds per CI, and the several pushes would have flush that out I think.